TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TwoControllerDriverControlled.java

```diff
@@ -74,40 +65,42 @@ public void runOpMode() throws InterruptedException {
                 y = applyJoystickSmoothing(y, joystickSmoothingExp);
                 r = applyJoystickSmoothing(r, joystickSmoothingExp);
 
-                //Applies calculated movement for both field-centric and not
-                if (fieldCentric) {
-                    //robot.getHeading() may be partially or not at all functional. Good luck, traveler.
-                    robot.drive.calculateDirectionsFieldCentric(x, y, r, robot.getHeading());
+                //employs gyro stabilization to linear driving
+                if (Math.abs(gamepad1.right_stick_x) < 0.05) {
+                    if (gyroAssistDebouncer.isPressed(Math.abs(gamepad1.right_stick_y) < 0.05)) {
+                        //identifies the desired heading angle to hold. This should only happen once per period of time that the right joystick is < 0.05
+                        target = robot.imu.getRobotYawPitchRollAngles().getYaw(AngleUnit.DEGREES);
+                    }
+                    robot.drive.calcGyroStabilized(x, y, target);
                 }
                 else {
-                    //applies drive values. Notice the negative R.
                     robot.drive.calculateDirections(x, y, r);
                 }
-                //Toggle field centric mode
-                //First implementation of new debouncer. If this fails at competition just delete.
-                if (dx.isPressed(gamepad1.triangle || gamepad2.x)) {
-                    fieldCentric = !fieldCentric;
-                }
-
-                //180
-                if (turnAroundDebouncer.isPressed(gamepad1.b)) {
-                    robot.drive.rotateLeftEncoder(180);
+                //90 degree turn left/right
+                if (gamepad1.left_trigger > 0.5) {
+                    if (turnAroundDebouncer.isPressed(gamepad1.left_trigger > 0.5)){
+                        target = robot.imu.getRobotYawPitchRollAngles().getYaw(AngleUnit.DEGREES) - 90;
+                    }
+                    robot.drive.calcGyroStabilized(0, 0, target);
```

Comment:
this could be changed to a `robot.drive.turn(90)` or `-90`. But I understand not wanting to add that

Reply:
Good idea. For some reason I think those might not work so I'll wait and see, if they do I'll fix that